### ISSUE TITLE
Support more lsp features

### ISSUE THREAD

**Issue by aszenz** [kaermorchen/twig-language-server#19]
Thanks for making this and improving the lives of twig developers.

Before using the vscode plugin I used https://marketplace.visualstudio.com/items?itemName=mblode.twig-language-2.

I'm missing some feature from it though:

1. Twig snippets  like these
```
do                {% do ... %}
extends           {% extends 'template' %}
from              {% from 'template' import 'macro' %}
```
2. Support formatting maybe integrate with third party plugin like prettydiff
3. Some form of go-to-definition (maybe for blocks and macros initially)

**Comment by kaermorchen** [comment]
Did you try to use the both extension together? They  shouldn't conflict.

**Comment by aszenz** [comment]
> Did you try to use the both extensions together? They shouldn't conflict.

Yes i could use the snippets from the library, and use this library for syntax highlighting

**Comment by Shifu462** [comment]
@aszenz 

> Some form of go-to definition

Implemented that here for `include('template.html.twig')` and other template inclusion statements/functions.
https://github.com/kaermorchen/twig-language-server/pull/27

Please tell me if something is missing for this feature. PR review is welcomed.
